tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -72,6 +76,36 @@ def get_gitlab_repo(repo='DataDog/datadog-agent', token=None) -> Project:
     return repo
 
 
+def refresh_pipeline(pipeline: ProjectPipeline, max_retries=2, retry_delay=1):
```

Comment:
I think using a decorator (either custom or with something like [tenacity](https://tenacity.readthedocs.io/en/latest/)) on the function you want to retry would be more generic and easier to maintain. What do you think?

Reply:
+1000 on this !

Reply:
+ the traceback logic you started to implement is really easily handled in the __exit__ function of a context manager